File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/contactAct/adapterRecV.java
package com.example.adminapp.contactAct;

import android.media.Image;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.adminapp.R;

import java.util.ArrayList;

public class adapterRecV extends RecyclerView.Adapter<adapterRecV.holder> {

    ArrayList<dataModel> dataHolder;
    ArrayList<dataModel> dataHolderBackup;

    public  void setFilteredList(ArrayList<dataModel> filteredList){
        this.dataHolder = (ArrayList<dataModel>) filteredList;
        notifyDataSetChanged();
    }

    public adapterRecV(ArrayList<dataModel> dataHolder) {
        this.dataHolder = dataHolder;
        dataHolderBackup = new ArrayList<>(dataHolder);
    }

    @NonNull
    @Override
    public adapterRecV.holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.single_row_design_contact_act,parent,false);
        return new holder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull adapterRecV.holder holder, int position) {
        holder.tvContactName.setText(dataHolder.get(position).getContactName());
        holder.tvContactNo.setText(dataHolder.get(position).getNumber());
        holder.img.setImageResource(dataHolder.get(position).getImg());
    }

    @Override
    public int getItemCount() {
        return dataHolder.size();
    }

    public class holder extends RecyclerView.ViewHolder {

        ImageView img;
        TextView tvContactName,tvContactNo;

        public holder(@NonNull View itemView) {
            super(itemView);

            img = itemView.findViewById(R.id.iVPersonContactAct);
            tvContactName = itemView.findViewById(R.id.tVPNameContactAct);
            tvContactNo = itemView.findViewById(R.id.tVPNoContactAct);
        }
    }
}

File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/contactAct/adapterRecV.kt
package com.example.adminapp.contactAct

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.adminapp.R
import com.example.adminapp.contactAct.MVCARCHITECTURECONTACTACT.dataModel

class adapterRecV(var dataHolder: ArrayList<dataModel>?) :
    RecyclerView.Adapter<adapterRecV.holder>() {
    var dataHolderBackup: ArrayList<dataModel>
    fun setFilteredList(filteredList: ArrayList<dataModel>?) {
        dataHolder = filteredList
        notifyDataSetChanged()
    }

    init {
        dataHolderBackup = ArrayList(
            dataHolder
        )
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): holder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.single_row_design_contact_act, parent, false)
        return holder(view)
    }

    override fun onBindViewHolder(holder: holder, position: Int) {
        holder.tvContactName.text = dataHolder!![position].contactName
        holder.tvContactNo.text = dataHolder!![position].number
        holder.img.setImageResource(dataHolder!![position].img)
    }

    override fun getItemCount(): Int {
        return dataHolder!!.size
    }

    inner class holder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var img: ImageView
        var tvContactName: TextView
        var tvContactNo: TextView

        init {
            img = itemView.findViewById(R.id.iVPersonContactAct)
            tvContactName = itemView.findViewById(R.id.tVPNameContactAct)
            tvContactNo = itemView.findViewById(R.id.tVPNoContactAct)
        }
    }
}
